docs/docs/how-to/installing-shadcn.md

```diff
@@ -0,0 +1,346 @@
+If you're comparing this file against the tailwind.config.js file in [ShadCN's documentation](https://ui.shadcn.com/docs/installation/manual), there are a few key differences:
+- I removed the references to `fontFamily`.
+- the `content` path is different. I've updated it specifically for Redwood.
+:::
+
+6. Configure styles
+
+Redwood already set up a CSS file for Tailwind: `web/src/index.css`.
+
+If you open the file, you should see Tailwind's base styles already listed:
+
+```css
+/**
+ * START --- SETUP TAILWINDCSS EDIT
+ *
+ * `yarn rw setup ui tailwindcss` placed these directives here
+ * to inject Tailwind's styles into your CSS.
+ * For more information, see: https://tailwindcss.com/docs/installation
+ */
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+/**
+ * END --- SETUP TAILWINDCSS EDIT
+ */
+ ```
+
+Add the following below the Tailwind setup:
+
+```
+@layer base {
+  :root {
+    --background: 0 0% 100%;
+    --foreground: 222.2 47.4% 11.2%;
+ 
+    --muted: 210 40% 96.1%;
+    --muted-foreground: 215.4 16.3% 46.9%;
+ 
+    --popover: 0 0% 100%;
+    --popover-foreground: 222.2 47.4% 11.2%;
+ 
+    --border: 214.3 31.8% 91.4%;
+    --input: 214.3 31.8% 91.4%;
+ 
+    --card: 0 0% 100%;
+    --card-foreground: 222.2 47.4% 11.2%;
+ 
+    --primary: 222.2 47.4% 11.2%;
+    --primary-foreground: 210 40% 98%;
+ 
+    --secondary: 210 40% 96.1%;
+    --secondary-foreground: 222.2 47.4% 11.2%;
+ 
+    --accent: 210 40% 96.1%;
+    --accent-foreground: 222.2 47.4% 11.2%;
+ 
+    --destructive: 0 100% 50%;
+    --destructive-foreground: 210 40% 98%;
+ 
+    --ring: 215 20.2% 65.1%;
+ 
+    --radius: 0.5rem;
+  }
+ 
+  .dark {
+    --background: 224 71% 4%;
+    --foreground: 213 31% 91%;
+ 
+    --muted: 223 47% 11%;
+    --muted-foreground: 215.4 16.3% 56.9%;
+ 
+    --accent: 216 34% 17%;
+    --accent-foreground: 210 40% 98%;
+ 
+    --popover: 224 71% 4%;
+    --popover-foreground: 215 20.2% 65.1%;
+ 
+    --border: 216 34% 17%;
+    --input: 216 34% 17%;
+ 
+    --card: 224 71% 4%;
+    --card-foreground: 213 31% 91%;
+ 
+    --primary: 210 40% 98%;
+    --primary-foreground: 222.2 47.4% 1.2%;
+ 
+    --secondary: 222.2 47.4% 11.2%;
+    --secondary-foreground: 210 40% 98%;
+ 
+    --destructive: 0 63% 31%;
+    --destructive-foreground: 210 40% 98%;
+ 
+    --ring: 216 34% 17%;
+ 
+    --radius: 0.5rem;
+  }
+}
+ 
+@layer base {
+  * {
+    @apply border-border;
+  }
+  body {
+    @apply bg-background text-foreground;
+    font-feature-settings: "rlig" 1, "calt" 1;
+  }
+}
+```
+
+7. Add a cn helper
+
+Shad uses a `cn` helper to make it easier to conditionally add Tailwind CSS classes.
+
+Within the `web/src` directory, create a folder called `lib`, with a file inside called `utils.ts`
+
+Add the following code:
+```ts
+// web/src/lib/utils.ts
+import { clsx, type ClassValue } from 'clsx'
+import { twMerge } from 'tailwind-merge'
+
+export function cn(...inputs: ClassValue[]) {
+  return twMerge(clsx(inputs))
+}
+```
```

Comment:
When I run e.g. `yarn shad card`

```bash
✔ Checking registry.
✔ Created 1 file:
  - src/components/ui/card.tsx
```
 
The component gets created with the import as follows:

```
import { cn } from "s/lib/utils"
```

Should be "src/lib/utils" ?

Reply:
@starfish456 Have you tried our shad setup command? https://github.com/redwoodjs/setup-rw-shad

Reply:
Oh nice, thanks!! I wasn't aware of that. Should this be documented?

Reply:
Yes! 🙂 Last I spoke to Amy about it we said we wanted to have both options documented. So people can choose to do it manually or just use the setup command. Obviously I'm partial to the setup command, since I wrote it 😆 So the more community input we can get on the different options the better!

Reply:
Awesome! I have made a few small improvements for the actual `yarn rw-shad` command that you'll use to add your components. I'll try to get those published in a bit.
Things should already work just fine. I've just improved how it handles already added components 🙂

Reply:
@starfish456 `rw-shad` was just updated to v1.1.0 please upgrade (just bump the version in the root `package.json` and then run `yarn install`)

And then let us know how you get on with adding various shadcn components!

Reply:
Hey @Tobbe 👋, a heads up that after using this setup for shadcn I get prerender errors, I think because the prerender server can't resolve the @ alias which is the default for shadcn components.

Have paths set correctly in .jsconfig and alias set in vite.config

Workaround for now is to use `src/components` instead of `@/components` for components in pages I'm prerendering

```
---------- Error rendering path "/" ----------
Error: Cannot find module '@/components/ui/button'
Require stack:
- /Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/web/src/components/Public/Landing/LandingNav.js
- /Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/web/src/layouts/AppLayout/AppLayout.jsx
- /Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/web/src/Routes.jsx
- /Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/web/src/App.jsx
- /Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/@redwoodjs/prerender/dist/runPrerender.js
- /Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/@redwoodjs/prerender/dist/index.js
- /Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/@redwoodjs/cli/dist/commands/prerenderHandler.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1048:15)
    at Function.Module._load (node:internal/modules/cjs/loader:901:27)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at Object.<anonymous> (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/web/src/components/Public/Landing/LandingNav.js:5:1)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._compile (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/pirates/lib/index.js:117:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Object.newLoader [as .js] (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/pirates/lib/index.js:121:7)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Function.Module._load (node:internal/modules/cjs/loader:938:12)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at Object.<anonymous> (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/web/src/layouts/AppLayout/AppLayout.jsx:3:1)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._compile (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/pirates/lib/index.js:117:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Object.newLoader [as .jsx] (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/pirates/lib/index.js:121:7)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Function.Module._load (node:internal/modules/cjs/loader:938:12)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at Object.<anonymous> (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/web/src/Routes.jsx:4:1)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._compile (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/pirates/lib/index.js:117:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Object.newLoader [as .jsx] (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/pirates/lib/index.js:121:7)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Function.Module._load (node:internal/modules/cjs/loader:938:12)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at Object.<anonymous> (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/web/src/App.jsx:5:1)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._compile (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/pirates/lib/index.js:117:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Object.newLoader [as .jsx] (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/pirates/lib/index.js:121:7)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Function.Module._load (node:internal/modules/cjs/loader:938:12)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at runPrerender (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/@redwoodjs/prerender/dist/runPrerender.js:256:7)
    at prerenderRoute (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/@redwoodjs/cli/dist/commands/prerenderHandler.js:237:29)
    at _Task.task [as taskFn] (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/@redwoodjs/cli/dist/commands/prerenderHandler.js:168:11)
    at _Task.run (/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/node_modules/listr2/dist/index.cjs:2049:11)

[FAILED] Failed to render "/Users/hamishirving/Documents/Personal/Projects/Stapha/web-app/web/src/pages/Public/HomePage/HomePage.jsx"
```

Reply:
Hi @hamishirving nice to hear from you :)

> after using this setup for shadcn

"this setup" as the manual setup described in this PR, or the https://github.com/redwoodjs/setup-rw-shad command I recommended earlier in this comment thread?

Reply:
@Tobbe you too, you've been busy with rw!

I used the manual setup in this PR, haven't tried your command yet, does that support prerendering (and I saw another comment about tests)?

Reply:
My command uses `src/`, just to try to keep things feeling Redwoody even if it comes from another project. So I haven't explicitly tried with prerendering. But given that I don't use any special aliases I hope that it works. And if it doesn't I'll make sure to fix that :)

Reply:
And yeah, I expect the same thing to apply to tests as well

Reply:
@Tobbe  Just a heads up - `yarn dlx setup-rw-shad` fails with:
`TypeError: yargs.scriptName is not a function at file:///...node_modules/setup-rw-shad/dist/cli.js`

I tried different yarn versions but got the same error. The only way I got it working was by:
1. Cloning the repository
2. Running `yarn install`
3. Running the CLI locally

Not sure if I missed something, but wanted to let you know in case others run into the same issue!

Reply:
@ghkich Thanks! Fixed now